source/packet.cpp

```diff
@@ -4,6 +4,18 @@ namespace sim {
 Packet::Packet(PacketType a_type, std::uint32_t a_size, IFlow* a_flow)
     : type(a_type), size(a_size), flow(a_flow) {}
 
+std::shared_ptr<IRoutingDevice> Packet::get_destination() const {
```

Comment:
Please, check flow for nullptr:
```suggestion
std::shared_ptr<IRoutingDevice> Packet::get_destination() const {
    if (flow == nullptr) {
        return nullptr;
    }
```